### DISCUSSION TITLE
How to setup `native` on Zen Browser?

### DISCUSSION THREAD

**Discussion by azzamsa** [tridactyl/tridactyl#5094] [Q&A] [answered]
Hi. @caputchinefrobles 

Would you like to elaborate more regarding your comments here https://github.com/tridactyl/tridactyl/issues/5034#issuecomment-2335017204?

1. flatpak install flathub io.github.zen_browser.zen
2. Install https://addons.mozilla.org/en-US/firefox/addon/tridactyl-vim/
3. Run `curl -fsSl https://raw.githubusercontent.com/tridactyl/native_messenger/master/installers/install.sh -o /tmp/trinativeinstall.sh && sh /tmp/trinativeinstall.sh 1.24.1` (taken from `:nativeinstall`)
4. Run `flatpak permission-set webextensions tridactyl io.github.zen_browser.zen yes`

What to do next?

My profile is in `~/.var/app/io.github.zen_browser.zen/.zen/kmh7v15u.Default (alpha)` and the command from step 3 put the file in `~/.mozilla/native-messaging-hosts`. Should I move those files? 

Thanks!

**Comment by caputchinefrobles** [comment]
Hi @azzamsa. 

The referenced commend does not have to do with setting up native on zen browser. I suppose it won't help much on your case. 

Regarding the native install, you could try running `:native` to check if it is installed correctly.

**Comment by bovine3dom** [comment, marked as answer]
I'm not sure Zen supports native messengers yet, see https://github.com/zen-browser/desktop/issues/725

Once it does you'd probably need to make a symlink as suggested here https://github.com/tridactyl/tridactyl/issues/3965

**Reply by azzamsa** [reply]
Ah, thanks!  

TIL. I assumed Zen wouldn’t need to implement that since I thought it only modified the CSS in Firefox.  
I appreciate the link to the related issue!

**Comment by HendrikSP** [comment]
On macOS i needed to run this:
```sh
ln -sf ../../Mozilla/NativeMessagingHosts/tridactyl.json $HOME/Library/Application\ Support/zen/NativeMessagingHosts/tridactyl.json
```
